### ISSUE TITLE
Undefined header in config doesn't cause test suite to fail

### ISSUE THREAD

**Issue by simonplend** [Financial-Times/n-test#63]
# Expected behaviour

A test suite should fail if the config in `smoke.js` is invalid.

# Actual behaviour

If a header specified in `smoke.js` config is `null` or `undefined`, a `TypeError` will be thrown, however the current error handling behaviour will log this error out and continue, meaning that the test suite won't fail even if the config is invalid.

# Background

> fyi, our next-control-centre build passed - https://circleci.com/gh/Financial-Times/next-control-centre/560 - but threw a non-fatal error from the smoke tests:
```TypeError: Cannot read property 'toString' of undefined
at PuppeteerPage.init (/home/ubuntu/next-control-centre/node_modules/@financial-times/n-test/lib/smoke/puppeteer-page.js:72:64)
```

Source: [Slack](https://financialtimes.slack.com/archives/C9NR0J683/p1527078338000330)

The `TEST_SESSION_ID` environment variable referenced in this project's [smoke.js](https://github.com/Financial-Times/next-control-centre/blob/master/test/smoke.js#L6) is not defined for that specific build, so the value for the header in `smoke.js` was `undefined`.

**Comment by simonplend** [comment]
@jonnywyatt Thought you might like to know I finally fixed this. `n-test` will now catch invalid headers in config (i.e. `undefined` or `null`). In the near-ish future I'm planning to add further validation of all test config before we even attempt to run tests.

**Comment by simonplend** [comment]
Fixed in PR #65

**Comment by jonnywyatt** [comment]
Great stuff thanks!
